sources/vcdMaker/src/vcdMaker.cpp

```diff
@@ -45,15 +46,36 @@ int main(int argc, const char *argv[])
     // Source registry.
     SIGNAL::SourceRegistry registry;
 
+    // Line counter.
+    INSTRUMENT::LineCounter *lineCounter = NULL;
+
     try
     {
-        // Parse the log file.
+        // Create the log parser.
         PARSER::TxtParser txtLog(cli.GetInputFileName(),
                                  cli.GetTimebase(),
                                  registry,
-                                 cli.GetLineCounterName(),
                                  cli.IsVerboseMode());
 
+        if (cli.GetLineCounterName().size())
+        {
+            // Register the line counting instrument.
+            lineCounter = new INSTRUMENT::LineCounter(cli.GetInputFileName(),
```

Comment:
It has been decided to leave it as it is. Containers store just objects. They cannot store references. So, in our case they store pointers. The reference is passed so as not to check against the NULL pointer. It should always be not NULL. The solution with std::ref worked but the syntax was confusing.